TLSharp.Core/TelegramClient.cs

```diff
@@ -105,11 +106,63 @@ public bool IsUserAuthorized()
             if (_sender == null)
                 throw new InvalidOperationException("Not connected!");
 
-            var authCheckPhoneRequest = new TLRequestCheckPhone() { phone_number = phoneNumber };
-            await _sender.Send(authCheckPhoneRequest);
-            await _sender.Receive(authCheckPhoneRequest);
+            var request = new TLRequestCheckPhone() { phone_number = phoneNumber };
 
-            return authCheckPhoneRequest.Response.phone_registered;
+            var completed = false;
+            while (!completed)
+            {
+                request = new TLRequestCheckPhone() { phone_number = phoneNumber };
+                try
+                {
+                    await _sender.Send(request);
+                    await _sender.Receive(request);
+
+                    completed = true;
+                }
+                catch (PhoneMigrationException ex)
+                {
+                    await ReconnectToDcAsync(ex.DC);
+                }
+            }
+
+            return request.Response.phone_registered;
+        }
+
+        public async Task<TLAbsChannelDifference> GetFirst100HundredUpdatesFromTheChannel(TLChannel channel)
```

Comment:
This part of the commit is not explained in your commit message, it appears you need to split the commit into two commits.